webui/src/pages/repositories/repository/pulls/pullDetails.jsx

```diff
@@ -58,17 +96,33 @@ const PullDetailsContent = ({repo, pull}) => {
                     Opened on {createdAt.format("MMM D, YYYY")} ({createdAt.fromNow()}).
                 </Card.Header>
                 <Card.Body className="description">
-                    {pull.description}
+                    <Markdown>{pull.description}</Markdown>
                 </Card.Body>
             </Card>
             <div className="bottom-buttons-row mt-4 clearfix">
+                {error && <AlertError error={error} onDismiss={() => setError(null)}/>}
                 <div className="bottom-buttons-group float-end">
-                    <Button variant="outline-secondary" className="text-secondary-emphasis me-2">
-                        Close pull request
-                    </Button>
-                    <Button variant="success">
-                        <GitMergeIcon/> Merge pull request
-                    </Button>
+                    {pull.status !== PullStatus.merged &&
+                        <Button variant="outline-secondary"
+                                className="text-secondary-emphasis me-2"
+                                disabled={loading}
+                                onClick={changePullStatus(isPullOpen() ? PullStatus.closed : PullStatus.open)}>
+                            {loading ?
+                                <span className="spinner-border spinner-border-sm text-light" role="status"/> :
+                                <>{isPullOpen() ? "Close" : "Re-open"} pull request</>
+                            }
+                        </Button>
+                    }
+                    {isPullOpen() &&
```

Comment:
This condition is not enough - Merge should be also disabled if no changes (due to modifications of the branches, or if any of them no longer exists)

Reply:
True - 
Will be handled in a separate PR (or deferred).